swap_meet/vendor.py

```diff
@@ -1,2 +1,75 @@
+#Wave 1
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item 
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        return False
+        
+#Wave2
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+
+        return None
+
+#Wave3
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            return True
+        
+        return False
+        
+#Wave4
+    def swap_first_item(self, other_vendor):
+        if self.inventory and other_vendor.inventory:
+            self.swap_items(
+                other_vendor, 
+                self.inventory[0], 
+                other_vendor.inventory[0]
+            )
+            return True
+
+        return False
+    
+#Wave6
+    def get_by_category(self, category):
+        inv = self.inventory.copy()
+        return list(filter(lambda item: item.get_category() == category, inv))
```

Comment:
Great use of lambdas and the filter function here! One small change would be that the filter function returns a new list, so you don't technically have to make a copy of self.inventory! You could use it as is within your filter function!

Reply:
Good point, I removed the .copy()